Explorer/Assets/DCL/SDKComponents/AvatarAttach/Systems/AvatarAttachHandlerSystem.cs

```diff
@@ -24,11 +24,12 @@ public partial class AvatarAttachHandlerSystem : BaseUnityLoopSystem, IFinalizeW
     {
         private static readonly QueryDescription ENTITY_DESTRUCTION_QUERY = new QueryDescription().WithAll<DeleteEntityIntention, AvatarAttachComponent>();
         private static readonly QueryDescription COMPONENT_REMOVAL_QUERY = new QueryDescription().WithAll<AvatarAttachComponent>().WithNone<DeleteEntityIntention, PBAvatarAttach>();
-        private AvatarBase ownPlayerAvatarBase;
+        private static AvatarBase mainPlayerAvatarBase;
```

Comment:
can we keep it not static?

Reply:
Uhmmm but if it's not static we would have a different `mainPlayerAvatarBase` for each scene (thus repeating [the query that fills that variable](https://github.com/decentraland/unity-explorer/blob/9bd7f7a34a608b4a336e39e256c4063a0d5feb4b/Explorer/Assets/DCL/SDKComponents/AvatarAttach/Systems/AvatarAttachHandlerSystem.cs#L32), for each scene) right? since Systems get instantiated during the `InjectToWorld()` of their plugin right?

That's why I made it static, to have only 1 reference there, and thus only the first scene that instantiates the `AvatarAttachHandlerSystem` would run [the globalWorld query](https://github.com/decentraland/unity-explorer/blob/9bd7f7a34a608b4a336e39e256c4063a0d5feb4b/Explorer/Assets/DCL/SDKComponents/AvatarAttach/Systems/AvatarAttachHandlerSystem.cs#L32) only once and no other scene would run that query.

Reply:
ah, ok. Didn't see it. If it could me moved in some upper level, where it could be initialized once, then it would be nice. Otherwise, you can keep it.